### PR TITLE
[release-1.21] - Backport Fix storing bootstrap data with empty token string

### PR DESCRIPTION
**Pull request by galal-hussein**
#### Linked Issues ####

* https://github.com/rancherlabs/rancher-security/issues/831
* #3629
* #3632

### CODE REVIEW

**Review by briandowns — APPROVED** [review]

**Comment by brandond** [comment]
Will need https://github.com/k3s-io/k3s/pull/3545 rolled in as well.

**Review by brandond — CHANGES_REQUESTED** [review]
Will need https://github.com/k3s-io/k3s/pull/3545 rolled in as well.

**Review by briandowns — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -85,7 +85,7 @@ require (
 	github.com/gorilla/mux v1.8.0
 	github.com/gorilla/websocket v1.4.2
 	github.com/k3s-io/helm-controller v0.10.1
-	github.com/k3s-io/kine v0.6.0
+	github.com/k3s-io/kine v0.6.1
```
Comment (brandond) [inline go.mod:88]: Need to bump to https://github.com/k3s-io/kine/releases/tag/v0.6.2 for the delete fix

**Review by brandond — CHANGES_REQUESTED** [review]

**Comment by galal-hussein** [comment]
https://github.com/k3s-io/k3s/issues/3629 adding issue for backporting kine

**Review by brandond — APPROVED** [review]

**Comment by galal-hussein** [comment]
https://github.com/k3s-io/k3s/issues/3632 - backport issue for empty node password file

**Pull request merged**